src/TwitterDB.js

```diff
@@ -0,0 +1,462 @@
+        reject(err);
+      }
+      if(row !== undefined){
+        resolve(row.room_id);
+      }
+      else {
+        resolve(null);
+      }
+    });
+  });
+}
+
+TwitterDB.prototype.set_timeline_room = function(user_id,room_id){
+  this.db.run(
+    `
+    REPLACE INTO timeline_room
+    VALUES ($user_id,$room_id)
+    `
+  ,{
+    $user_id:user_id,
+    $room_id:room_id
+  },(err) =>{
+    if(err != null){
+      log.error("TwitDB","Error storing timeline room for user: %s",err);
+    }
+  });
+}
+
+TwitterDB.prototype.remove_timeline_room = function(user_id){
+  this.db.run(
+    `
+    DELETE FROM  timeline_room
+    WHERE timeline_room.user_id = $user_id;
+    `
+  ,{
+    $user_id:user_id
+  },(err) =>{
+    if(err != null){
+      log.error("TwitDB","Error deleting timeline room for user: %s",err);
+    }
+  });
+}
+
+TwitterDB.prototype.remove_client_data = function(user_id){
+  this.db.run(
+    `
+    DELETE FROM twitter_account
+    WHERE twitter_account.user_id = $user_id;
+    `
+  ,{
+    $user_id:user_id
+  },(err) =>{
+    if(err != null){
+      log.error("TwitDB","Error deleting client data for user: %s",err);
+    }
+  });
+}
+
+TwitterDB.prototype.get_dm_room = function(users){
+  log.info("TwitDB","Retrieving dm room: %s",users);
+  return new Promise((resolve,reject) =>{
+    this.db.get(
+      `
+      SELECT room_id
+      FROM dm_room
+      WHERE dm_room.users = $users;
+      `
+    ,{
+      $users: users
+    }
+    ,(err,row) =>{
+      if(err != null){
+        log.error("TwitDB","Error retrieving dm room: %s",err.Error);
+        reject(err);
+      }
+      if(row !== undefined){
+        resolve(row.room_id);
+      }
+      else {
+        resolve(null);
+      }
+    });
+  });
+}
+
+TwitterDB.prototype.get_users_from_dm_room = function(room_id){
+  log.info("TwitDB","Retrieving dm room: %s",room_id);
+  return new Promise((resolve,reject) =>{
+    this.db.get(
+      `
+      SELECT users
+      FROM dm_room
+      WHERE dm_room.room_id = $room_id;
+      `
+    ,{
+      $room_id: room_id
+    }
+    ,(err,row) =>{
+      if(err != null){
+        log.error("TwitDB","Error retrieving dm room: %s",err.Error);
+        reject(err);
+      }
+      if(row !== undefined){
+        resolve(row.users);
+      }
+      else {
+        resolve(null);
+      }
+    });
+  });
+}
+
+TwitterDB.prototype.add_dm_room = function(room_id,users){
+  log.info("TwitDB","Storing dm room %s",room_id);
+  this.db.run(
+    `
+    INSERT INTO dm_room VALUES ($room_id,$users);
+    `
+  ,{
+    $room_id: room_id,
+    $users: users
+  },
+  function (err) {
+    if(err){
+      log.error("TwitDB","Error storing dm room: %s",err);
+      return;
+    }
+    log.info("TwitDB","Stored dm room %s",room_id);
+  });
+}
+
+//Caches every user profile we grab from Twitter so as to not go over our limits.
+TwitterDB.prototype._create_profile_cache = function(){
```

Comment:
All of these `_create_xxx` functions would benefit from being a single `_create` function which accept an array of strings which contain the `CREATE TABLE` statements. This means that:
 - All your CREATE TABLE stuff is in one place rather than spread over many functions.
 - You don't repeat yourself when handling errors.